Charon/filetypes/GCodeSocket.py

```diff
@@ -0,0 +1,82 @@
+# Copyright (c) 2018 Ultimaker B.V.
+# libCharon is released under the terms of the LGPLv3 or higher.
+#
+# This class is used to read GCode stream that are served
+# dynamically over a TCP connection.
+
+import socket
+import struct
+from io import BytesIO, SEEK_SET, SEEK_CUR
+
+from typing import Any, Dict, IO, Optional, List
+
+from Charon.filetypes.GCodeFile import GCodeFile
+from urllib.parse import urlparse
+
+
+class SocketFileStream(BytesIO):
+    def __init__(self, sock_object: socket.socket):
```

Comment:
```suggestion
    def __init__(self, sock_object: socket.socket) -> None:
```
If you don't fully type a function, mypy won't check it (it's weird like that)

Reply:
funky I didn't explicitly check for typing errors because I would expect our CI street to pick up on those? @nallath

Reply:
Oh, it's not an error. It basicly won't check the content of the function for typing if you don't fully type the signature

So this:
```python3
def __init__(self, sock_object: socket.socket):
    self.bla: bool = True
    self.bla = "haha" 
```
will not cause typing errors

this will:
```python3
def __init__(self, sock_object: socket.socket) -> None:
    self.bla: bool = True
    self.bla = "haha" 
```

Reply:
```
disallow_untyped_calls = False
disallow_untyped_defs = False
disallow_incomplete_defs = False
```
probably one of these in `mypy.ini` causing this behaviour